python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_categorical.py

```diff
@@ -0,0 +1,375 @@
+"""
+Kriging: metamodel with continuous and categorical variables
+============================================================
+"""
+# %%
+# We consider here the surrogate modeling of an analytical function characterized by
+# continuous and categorical variables
+#
+
+# %%
+import openturns as ot
+import openturns.experimental as otexp
+import numpy as np
+import matplotlib.pyplot as plt
+
+ot.RandomGenerator.SetSeed(5)
+
+# %%
+# We first show the advantage of modeling the various levels of a mixed
+# continuous / categorical function through a single surrogate model
+# on a simple test-case, defined below.
+
+
+# %%
+def illustrativeFunc(inp):
+    x, z = inp
+    x = 7 * x
+    y = np.cos(x) + 0.5 * z
+
+    return [y]
+
+
+dim = 2
+fun = ot.PythonFunction(dim, 1, illustrativeFunc)
+lvls = 2  # Number of discrete levels for z
+# Input distribution
+dist = ot.ComposedDistribution(
+    [ot.Uniform(0, 1), ot.UserDefined(np.arange(0, lvls).reshape((-1, 1)))]
+)
+
+# %%
+# In this example, we compare the performances of the LatentVariableModel with a naive
+# approach, which would consist in modeling each combination of categorical
+# variables through a separate and independent Gaussian process.
+
+# %%
+# In order to deal with mixed continuous / discrete problems we can rely on the
+# ProductCovarianceModel class. We start here by defining the product kernel,
+# which combines SquaredExponential kernels for the continuous variables, and
+# LatentVariableModel for the discrete ones.
+
+# %%
+latdim = 1  # Dimension of the latent space
```

Comment:
The variable name `latdim` hides its meaning. I suggest to name it `latentDimension`, so that the comment is unnecessary:
```suggestion
latentDimension = 1
```